agent/agent.go

```diff
@@ -82,11 +126,24 @@ func New(config *Config, auth authn.Authentication, provider attestation.Provide
 		transport.TLSClientConfig = tlsConfig
 	}
 
+	// Sort routes by priority (higher first)
+	routes := make([]RouteRule, len(config.Router.Routes))
+	copy(routes, config.Router.Routes)
+
+	for i := 0; i < len(routes); i++ {
+		for j := i + 1; j < len(routes); j++ {
+			if routes[j].Priority > routes[i].Priority {
+				routes[i], routes[j] = routes[j], routes[i]
+			}
+		}
+	}
```

Comment:
Let's use `sort.Sort` and create `ByPriority` as `sort.Interface` implementation. Check example https://pkg.go.dev/sort#example-package.